cmd/device.go

```diff
@@ -177,12 +180,18 @@ func sendDeviceHeartbeats(wg *sync.WaitGroup, beat *client.DeviceHeartbeat, wsm
 			}
 
 			// fallback to sending heartbeat to HTTP endpoint if there is an error with websocket
+			log.Error(err, "Falling back to HTTP heartbeats")
 
 		} else {
 			// device is not connected to an audio server
 
-			// sleep for heartbeat interval
-			time.Sleep(HeartbeatInterval * time.Second)
+			if firstHeartbeat {
```

Comment:
Instead of this firstHeartbeat, should we move lines from 203 to 210 to the beginning of the loop?

Reply:
but then i think you would always be sending http heartbeats, even when connected to the backend and sending heartbeats via websocket

Reply:
yeah, i guess there isn't really a good way to do it without reshuffling if statements.